src/flb_config.c

```diff
@@ -161,6 +161,8 @@ struct flb_config *flb_config_init()
     config->http_port    = flb_strdup(FLB_CONFIG_HTTP_PORT);
 #endif
 
+    config->http_proxy = getenv("HTTP_PROXY");
```

Comment:
usually when HTTP_PROXY="" (empty), proxy should be disabled. could you set it to NULL when it's empty string?

Reply:
It looks like `getenv` already returns `NULL` in this case:

```
/* Return the value of envariable NAME, or NULL if it doesn't exist.  */
extern char *getenv (const char *__name) __THROW __nonnull ((1)) __wur;
```

Reply:
Oh I see, you want to prevent the case where `HTTP_PROXY=""`. OK, fixing that now.

Reply:
Done.

Reply:
nit: could you squash the commits? having separete commits for small fixes is not clean.

Reply:
Sound good. (I did squash before in github but the previous review got lost). At this point it should not matter.